### PR TITLE
Add options.autoIncrement to MySQL

### PR DESCRIPTION
**Pull request by mdpauley**
The initial AUTO_INCREMENT value for the table.

### CODE REVIEW

**Comment by janmeier** [comment]
Needs a test, and the option should be added to the API docs for `sequelize.define`

**Comment by mdpauley** [comment]
Sorry about that! Added the test and to the API docs.

File: test/integration/dialects/mysql/query-generator.test.js [hunk 1]
```diff
@@ -106,6 +106,10 @@ if (Support.dialectIsMySQL()) {
         {
           arguments: ['myTable', {title: 'VARCHAR(255)', name: 'VARCHAR(255)'}, {uniqueKeys: [{fields: ['title', 'name']}]}],
           expectation: 'CREATE TABLE IF NOT EXISTS `myTable` (`title` VARCHAR(255), `name` VARCHAR(255), UNIQUE `uniq_myTable_title_name` (`title`, `name`)) ENGINE=InnoDB;'
+        },
+        {
+          arguments: ['myTable', {id: 'INTEGER auto_increment PRIMARY KEY'}, {initialAutoIncrement: 1000001}],
+          expectation: 'CREATE TABLE IF NOT EXISTS `myTable` (`id` INTEGER auto_increment , PRIMARY KEY (`id`)) ENGINE=InnoDB   AUTO_INCREMENT=1000001;'
```
Comment (mickhansen) [inline test/integration/dialects/mysql/query-generator.test.js]: Why the multiple spaces?

File: docs/api/sequelize.md [hunk 2]
```diff
@@ -482,6 +482,7 @@ For more about validation, see http://sequelizejs.com/docs/latest/models#validat
 | [options.charset] | String |  |
 | [options.comment] | String |  |
 | [options.collate] | String |  |
+| [options.initialAutoIncrement] | String | Set the initial AUTO_INCREMENT value for the table in MySQL. |
```
Comment (mickhansen) [inline docs/api/sequelize.md]: You have to add the docs to the code, `docs/api` is auto generated

File: test/integration/dialects/mysql/query-generator.test.js (hunk 1 above)
Comment (mdpauley) [inline test/integration/dialects/mysql/query-generator.test.js]: They are due to the spaces in [query variable](https://github.com/mdpauley/sequelize/blob/1afa4bb103d6db6ea9872657cc624d53814140a3/lib/dialects/mysql/query-generator.js#L32) between charset and collation that weren't set in that example. In my opinion, a proper fix would be to add the spaces in the [values object](https://github.com/mdpauley/sequelize/blob/1afa4bb103d6db6ea9872657cc624d53814140a3/lib/dialects/mysql/query-generator.js#L64-L72), like with comment, when charset and collation are generated.

Example:
```javascript
var query = 'CREATE TABLE IF NOT EXISTS <%= table %> (<%= attributes%>)<%= comment %> ENGINE=<%= engine %><%= charset %><%= collation %><%= initialAutoIncrement %>'
   , primaryKeys = []
   , foreignKeys = {}
   , attrStr = [];
```
and
```javascript
var values = {
  table: this.quoteTable(tableName),
  attributes: attrStr.join(', '),
  comment: options.comment && Utils._.isString(options.comment) ? ' COMMENT ' + this.escape(options.comment) : '',
  engine: options.engine,
  charset: (options.charset ? ' DEFAULT CHARSET=' + options.charset : ''),
  collation: (options.collate ? ' COLLATE ' + options.collate : ''),
  initialAutoIncrement: (options.initialAutoIncrement ? ' AUTO_INCREMENT=' + options.initialAutoIncrement : '')
}
```

I could make those changes and include with this pull request.

Comment (mickhansen) [inline test/integration/dialects/mysql/query-generator.test.js]: @mdpauley if you are up for it that would be welcome. But not critical for the merge of this PR.

**Comment by mdpauley** [comment]
@janmeier & @mickhansen everything should be up to standard now.

**Comment by janmeier** [comment]
Sweet, thanks @mdpauley  :)

**Pull request merged**